src/components/MenuItem/MenuItem.tsx

```diff
@@ -82,25 +83,61 @@ export const MenuItem: FC<MenuItemProps> = ({
 
     const textClass = menuItemTextColorRecord[style][textState];
 
+    const [mainElementType, setMainElementType] = useState('span');
+
+    useEffect(() => {
+        if (link || onClick) {
+            setMainElementType(getItemElementType(link, onClick));
+        }
+    }, [link, onClick]);
```

Comment:
Do we need this? Seems we could do with: 
`const elementType = getItemElementType(link, onClick)`

Reply:
Having the `useEffect` would avoid any unnecessary call to `getItemElementType`. Since it is dependent of the `link` and `onClick` props which are used only on the new implementation of `MenuItem`, a side effect hook here is the proper way to go, imho.